### ISSUE TITLE
Show name of Extensions wanting to make changes to the terminal's environment

### ISSUE THREAD

**Issue by sguillia** [microsoft/vscode#108940]
**TL;DR**: I'd like this basic security feature

> Extensions **microsoft.something** want to make the following changes to the terminal's environment

---

I saw this message appear in my terminal:
> Extensions want to make the following changes to the terminal's environment:
>
> GIT_ASKPASS=/usr/share/code-insiders/resources/app/extensions/git/dist/askpass.sh

I understand some extension is willing to intercept my git password.

In this case, the change is certainly legit. But how am I supposed to know if it's not from a malicious extension?

By seeing the extension name, I can guess if it makes sense for this extension to mess with my environment.

Even better, extensions could want to explain why they change the environment; I always feel strange when they do, especially since I never requested such a behavior.

Oh and- how about the ability to refuse the change?

```
Version: 1.51.0-insider
Commit: 89c002ab02f87102d91efc83c191ef1174756c6a
Date: 2020-10-16T05:12:37.273Z (3 days ago)
Electron: 9.3.2
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Linux x64 4.15.0-120-generic
```

**Comment by Tyriar** [comment]
> In this case, the change is certainly legit.

Yep, this is the builtin git extension.

> But how am I supposed to know if it's not from a malicious extension?

The current contract is that you installed the extension so you've given it permission to do its thing.

> Even better, extensions could want to explain why they change the environment; I always feel strange when they do, especially since I never requested such a behavior.

I considered this to begin with and received some feedback that the dialog was getting too complicated. Additionally if it happened to be from a malicious extension, they could just make something up anyway.

> Oh and- how about the ability to refuse the change?

Currently you would either disable the extension or turn it off via a setting if the extension has provided one.

---

Closing this out as a duplicate of https://github.com/microsoft/vscode/issues/96699 to show the extension name.
